pallets/parachain-staking/src/tests.rs

```diff
@@ -901,3 +905,39 @@ fn payouts_follow_nomination_changes() {
 		assert_eq!(events(), expected);
 	});
 }
+
+#[test]
+fn round_immediately_jumps_if_current_duration_exceeds_new_blocks_per_round() {
+	one_collator_two_nominators().execute_with(|| {
+		// Default round every 5 blocks, but MinBlocksPerRound is 3 and we set it to min 3 blocks
+		roll_to(8);
+		// chooses top TotalSelectedCandidates (5), in order
+		let init = vec![
+			Event::CollatorChosen(2, 1, 40),
+			Event::NewRound(5, 2, 1, 40),
+		];
+		assert_eq!(events(), init);
+		assert_ok!(Stake::set_blocks_per_round(Origin::root(), 3u32));
+		assert_eq!(
+			last_event(),
+			MetaEvent::stake(Event::SetBlocksPerRound(2, 5, 5, 3))
+		);
+		roll_to(9);
+		assert_eq!(last_event(), MetaEvent::stake(Event::NewRound(8, 3, 1, 40)));
+	});
+	one_collator_two_nominators().execute_with(|| {
```

Comment:
Is it common to have these both in the same `#[test]` function?

Reply:
I do that for the test of all of the genesis environments (`fn geneses()`). If it fits under the same test, I don't see a problem with it.